src/payform.coffee

```diff
@@ -123,28 +123,37 @@
       pattern: /^3[47]/
       format: /(\d{1,4})(\d{1,6})?(\d{1,5})?/
       length: [15]
-      cvcLength: [3..4]
+      cvcLength: [4]
+      luhn: true
+    }
+    # Must be above dinersclub.
+    {
+      type: 'hipercard'
+      pattern: /^(384100|384140|384160|606282|637095|637568|60(?!11))/
+      format: defaultFormat
+      length: [14..19]
+      cvcLength: [3]
       luhn: true
     }
     {
       type: 'dinersclub'
-      pattern: /^3[0689]/
-      format: /(\d{1,4})(\d{1,4})?(\d{1,4})?(\d{1,2})?/
+      pattern: /^(36|38|30[0-5])/
+      format: /(\d{1,4})(\d{1,6})?(\d{1,4})?/
       length: [14]
       cvcLength: [3]
       luhn: true
     }
     {
       type: 'discover'
-      pattern: /^6([045]|22)/
+      pattern: /^(6011|65|64[4-9]|622)/
```

Comment:
This pattern surfaces most BINs provided by Discover Network: https://www.discovernetwork.com/downloads/IPP_VAR_Compliance.pdf